### PR TITLE
[18.0][MIG] hr_employee_firstname: Migration to 18.0

### PR DESCRIPTION
**Pull request by edy1192**

### CODE REVIEW

**Comment by edy1192** [comment]
Hi @luistorresm @luisg123v could you check this, please?

Regards.

**Review by luistorresm — APPROVED** [review]

**Comment by andreagidaltig** [comment]
> Hi @luistorresm @luisg123v could you check this, please?
> 
> Regards.

LGTM :+1:

File: hr_employee_firstname/tests/test_hr_employee_firstname.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import odoo
 from odoo.exceptions import ValidationError
 from odoo.tests.common import TransactionCase
-from odoo.tools import submap
+from odoo.tools.misc import submap
```
Comment (luisg123v) [inline hr_employee_firstname/tests/test_hr_employee_firstname.py:8]: Why this change?

I couldn't find any related change that doesn't allow the previous syntax, and I think importing methods directly from `odoo.tools` is preferred.

If really required, it might be a good idea to document it in the commit description.

**Review by luisg123v — CHANGES_REQUESTED** [review]

**Comment by luisg123v** [comment]
Hi @legalsylvain, @pedrobaeza,

I'm not able to mark this PR as a migration one, despite being maintainer, see https://github.com/OCA/hr/pull/1397#issuecomment-2450588614.

Could it be https://github.com/OCA/oca-github-bot/issues/226 is not fully addressed?

CC @yajo

File: hr_employee_firstname/tests/test_hr_employee_firstname.py (hunk 1 above)
Comment (edy1192) [reply hr_employee_firstname/tests/test_hr_employee_firstname.py:8]: When I ran the tests to verify that they worked fine I got this error:

 ```
File "/home/odoo/instance/extra_addons/hr/hr_employee_firstname/tests/__init__.py", line 3, in <module>
    from . import test_hr_employee_firstname
  File "/home/odoo/instance/extra_addons/hr/hr_employee_firstname/tests/test_hr_employee_firstname.py", line 8, in <module>
    from odoo.tools import submap
ImportError: cannot import name 'submap' from 'odoo.tools' (/home/odoo/instance/odoo/odoo/tools/__init__.py)
```

I tried to import the tools library using `from odoo import tools` and then `tools.submap` but got the same error.

**Review by luisg123v — APPROVED** [review]
LGTM :+1:
